src/ui/public/state_management/state_storage/_tests__/hashed_item_store.js

```diff
@@ -0,0 +1,190 @@
+        hashedItemStore = new HashedItemStore(sessionStorage);
+      });
+
+      it('persists the item in sessionStorage', () => {
+        hashedItemStore.setItem(hash, item);
+        expect(sessionStorage.getItem(hash)).to.equal(item);
+      });
+
+      it('returns true', () => {
+        const result = hashedItemStore.setItem(hash, item);
+        expect(result).to.equal(true);
+      });
+    });
+
+    describe(`if the item doesn't exist in sessionStorage`, () => {
+      describe(`if there's storage space`, () => {
+        let sessionStorage;
+        let hashedItemStore;
+        const hash = 'a';
+        const item = JSON.stringify({});
+
+        beforeEach(() => {
+          sessionStorage = new StubBrowserStorage();
+          hashedItemStore = new HashedItemStore(sessionStorage);
+        });
+
+        it('persists the item in sessionStorage', () => {
+          hashedItemStore.setItem(hash, item);
+          expect(sessionStorage.getItem(hash)).to.equal(item);
+        });
+
+        it('returns true', () => {
+          const result = hashedItemStore.setItem(hash, item);
+          expect(result).to.equal(true);
+        });
+      });
+
+      describe(`if there isn't storage space`, () => {
+        let sessionStorage;
+        let hashedItemStore;
+        let storageSizeLimit;
+        const hash = 'a';
+        const item = JSON.stringify({});
+
+        beforeEach(done => {
+          sessionStorage = new StubBrowserStorage();
+          hashedItemStore = new HashedItemStore(sessionStorage);
+
+          // Add some items that will be removed.
+          hashedItemStore.setItem('1', item);
+          setTimeout(() => {
+            // Do this a little later so that this item is newer.
+            hashedItemStore.setItem('2', item);
+            // Cap the storage at its current size.
+            storageSizeLimit = sessionStorage.getStubbedSize();
+            sessionStorage.setStubbedSizeLimit(storageSizeLimit);
+            done();
+          }, 1);
+        });
+
+        describe('and the item will fit', () => {
+          it('removes older items until the new item fits', () => {
+            hashedItemStore.setItem(hash, item);
+            expect(sessionStorage.getItem('1')).to.equal(null);
+            expect(sessionStorage.getItem('2')).to.equal(item);
+          });
+
+          it('persists the item in sessionStorage', () => {
+            hashedItemStore.setItem(hash, item);
+            expect(sessionStorage.getItem(hash)).to.equal(item);
+          });
+
+          it('returns true', () => {
+            const result = hashedItemStore.setItem(hash, item);
+            expect(result).to.equal(true);
+          });
+        });
+
+        describe(`and the item won't fit`, () => {
+          let itemTooBigToFit;
+
+          beforeEach(() => {
+            // Make sure the item is longer than the storage size limit.
+            itemTooBigToFit = '';
+            const length = storageSizeLimit + 1;
```

Comment:
Teah, i don't think the `storageSizeLimit` or the `itemTooBigToFit` depend on a specific store instance. I don't think either needs to be in a `beforeEach`